confgenerator/confgenerator_test.go

```diff
@@ -105,91 +103,115 @@ func testGenerateConfsWithValidInput(t *testing.T, platform platformConfig) {
 		testName := d.Name()
 		t.Run(testName, func(t *testing.T) {
 			t.Parallel()
+			// Retrieve the expected golden conf files.
+			expectedFiles := readFileContents(t, testName, platform.OS, dir)
+
+			var got map[string]string
+
 			confDebugFolder := filepath.Join(dirPath, testName)
 			userSpecifiedConfPath := filepath.Join(confDebugFolder, "/input.yaml")
 			builtInConfPath := filepath.Join(confDebugFolder, "/built-in-config.yaml")
 			mergedConfPath := filepath.Join(confDebugFolder, "/merged-config.yaml")
 			if err = confgenerator.MergeConfFiles(userSpecifiedConfPath, confDebugFolder, platform.OS, apps.BuiltInConfStructs); err != nil {
-				t.Fatalf("MergeConfFiles(%q, %q) got: %v", userSpecifiedConfPath, confDebugFolder, err)
-			}
-
-			data, err := ioutil.ReadFile(mergedConfPath)
-			if err != nil {
-				t.Fatalf("ReadFile(%q) got: %v", userSpecifiedConfPath, err)
-			}
-			t.Logf("merged config:\n%s", data)
-			uc, err := confgenerator.ParseUnifiedConfigAndValidate(data, platform.OS)
-			if err != nil {
-				t.Fatalf("ParseUnifiedConfigAndValidate got: %v", err)
+				// TODO: Move this inside generateConfigs when we can do MergeConfFiles in-memory
+				if _, ok := expectedFiles["error"]; ok || *updateGolden {
+					got = map[string]string{
+						"error": err.Error(),
+					}
+				} else {
+					t.Fatalf("MergeConfFiles(%q, %q) got: %v", userSpecifiedConfPath, confDebugFolder, err)
+				}
 			}
 
-			// Retrieve the expected golden conf files.
-			expectedMainConfig := readFileContent(t, testName, platform.OS, goldenMainPath, true)
-			expectedParserConfig := readFileContent(t, testName, platform.OS, goldenParserPath, true)
+			defer func() {
+				// Ignore errors because we don't know if MergeConfFiles generated them.
```

Comment:
`MergeConfFiles` may have failed to generate the config files and returned an error, or it might have generated one or both of the config files and then returned an error. We don't know if it did generate any files if it returns an error.

The code does not leave the two files around if the test fails. It always removes them.

Reply:
Sent a suggested edit in https://github.com/GoogleCloudPlatform/ops-agent/pull/514#discussion_r843149562. Resolving this thread.